libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/NavigatorMapper.kt

```diff
@@ -125,7 +127,7 @@ internal class NavigatorMapper {
                         activeGuidanceInfo.routeProgress.fractionTraveled.toFloat()
                     )
 
-                    routeProgressBuilder.remainingWaypoints(legs.size - (legIndex + 1))
+                    routeProgressBuilder.remainingWaypoints(navigator.remainingWaypoints().size)
```

Comment:
It's possible to improve performance if you need only number of remaining waypoints from navigator. Use nextWaypointIndex from last NavigationStatus.

Reply:
Good point, thanks! 
Is nextWaypointIndex takes into account silent waypoints?

Reply:
Yes, it returns an index in an array of waypoints passed to setRoute, and we expect all endpoints there.